test/cpp_member_function.cpp

```diff
@@ -161,6 +161,71 @@ struct bar : foo<int>
     REQUIRE(count == 13u);
 }
 
+#if __cplusplus == 202002L
```

Comment:
Instead of a compile-time check, you need a runtime check based on the clang binary used by the parser, just as was done inside the code that added the compiler flags (as we might compile with an old compiler but new clang or vice-versa).